docs/configuration.rst

```diff
@@ -40,11 +40,12 @@ address.
 The ``WILDCARD_SENDERS`` setting is a comma delimited list of user email addresses
 that are allowed to send emails from any existing address (spoofing the sender).
 
-The ``AUTH_RATELIMIT_IP`` (default: 60/hour) holds a security setting for fighting
-attackers that waste server resources by trying to guess user passwords (typically
-using a password spraying attack). The value defines the limit of authentication
-attempts that will be processed on non-existing accounts for a specific IP subnet
-(as defined in ``AUTH_RATELIMIT_IP_V4_MASK`` and ``AUTH_RATELIMIT_IP_V6_MASK`` below).
+The ``AUTH_RATELIMIT_IP`` (default: 5/hour) holds a security setting for fighting
+attackers that attempt a password spraying attack. The value defines the limit of
+authentication attempts that will be processed on **distinct** non-existing
```

Comment:
Isn't this also checked for existing accounts?

Reply:
That's only checked for non-existing accounts. The other rate limit applies for existing accounts (XXX wrong attempts, regardless of source)

Reply:
Ah, yes. But we should also de-duplicate those.
The problem occuring when changing a password: lots of devices (not using a token) try to log-in using the old password which will get the source IP blocked, even though no-one is brute-forcing passwords.

Reply:
That's arguable... security wise it's not a great idea of storing a representation of attempted passwords (it may be near-misses); We could hash them but would have to store them unsalted.

The second ratelimit locks the account, not the IP address.

We have all kind of fancy ways of ensuring we don't block legitimate users: If a login is successful the IP address won't get into rate-limits; we also use a special type of cookies to ensure that devices that have already authenticated successfully get access even if the account is locked.

Reply:
IMHO the right thing to do is to ensure those devices use tokens. That may mean making the autodiscovery hand over tokens automatically.